public/app/percona/dbaas/components/Kubernetes/OperatorStatusItem/KubernetesOperatorStatus/UpdateOperatorModal/UpdateOperatorModal.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { FC, useCallback } from 'react';
+import { Button, HorizontalGroup, useStyles } from '@grafana/ui';
+import { Modal, logger } from '@percona/platform-core';
+import { Messages } from 'app/percona/dbaas/DBaaS.messages';
+import { UpdateOperatorModalProps } from './UpdateOperatorModal.types';
+import { getStyles } from './UpdateOperatorModal.styles';
+import { KubernetesService } from '../../../Kubernetes.service';
+
+const { title, confirm, cancel, buildUpdateOperatorMessage } = Messages.kubernetes.updateOperatorModal;
+
+export const UpdateOperatorModal: FC<UpdateOperatorModalProps> = ({
+  kubernetesClusterName,
+  selectedOperator,
+  isVisible,
+  setVisible,
+  setLoading,
+  setSelectedCluster,
+  setOperatorToUpdate,
+  onOperatorUpdated,
+}) => {
+  const styles = useStyles(getStyles);
+  const { operatorType, operatorTypeLabel, version, availableVersion } = selectedOperator;
+
+  const updateOperator = useCallback(async () => {
+    try {
+      setLoading(true);
+      onClose();
+
+      await KubernetesService.installOperator(kubernetesClusterName, operatorType, availableVersion as string);
+      onOperatorUpdated();
```

Comment:
Aren't we supposed to call `setLoading(false)` afterwards?

Reply:
No, because the list request is called right after. It works the same way with delete db clusters for example. I understand it's not obvious unless we look at `onOperatorUpdated` function but it avoids that annoying flicker in the UI by setting the loader to false and then immediately to true again  :D

Reply:
Yeah, I imagined that could be the case, but good to know 💪